### ISSUE TITLE
Feature Request - Alpine Images

### ISSUE THREAD

**Issue by wSedlacek** [railwayapp/nixpacks#729]
### Feature request

Currently the images output (at least for Node which is what I have tried) are ubuntu based.
I have seen on Dockerhub there are also Debian images.

I would like to build alpine images for my production node applications.

### Motivation

Alpine images are smaller (200MB compared to 600MB for the application I have tested with)
Alpine images have less security vulnerabilities.

Alpine:
![Screen Shot 2022-12-17 at 08 51 36](https://user-images.githubusercontent.com/8206108/208252606-0f36ab22-3d5e-4442-b276-e9c3de880fb6.png)

Nixpacks Ubuntu:
![Screen Shot 2022-12-17 at 08 51 43](https://user-images.githubusercontent.com/8206108/208252638-e5665679-6b78-4014-92f8-829b6ac26eeb.png)

Others have also shown interest: https://github.com/railwayapp/nixpacks/issues/419#issuecomment-1220756677

### Contribution

I would be happy to test any pre-release features to ensure stability.

**Comment by Milo123459** [comment]
This requires a lot of work due to the fact we support apt packages, and because Debian and ubuntu both support APT, it is far easier. Some providers also depend on apt packages, unfortunately, so migrating everything to APK just for that would be a huge hassle.

**Comment by wSedlacek** [comment]
Would [multi-stage](https://docs.docker.com/build/building/multi-stage/) builds help?
The ask here really is that the output image be alpine based.
As I work primarily with Node I am not aware of the scope of making alpine images for other applications.

**Comment by Milo123459** [comment]
I don't think so.

**Comment by aleksrutins** [comment]
I think this would be great! However, I don't think multi-stage builds would really work, because:
1. You still have to figure out how to get the packages into the Alpine-based run stage
2. Alpine uses musl instead of glibc, so things compiled in the build stage with a different distro wouldn't be able to run.

However, maybe this could be an option that makes it so that apt packages are simply not allowed; if you put it in bold red text and shout it at the user, people would hopefully take the necessary measures to get their apps working on Alpine. Also, a `NIXPACKS_APK_PACKAGES` variable would probably be a prudent addition.

Nixpacks was actually originally based on Alpine, since Alpine is what the official NixOS Docker image uses. However, it was switched over to Debian (and now Ubuntu) because folks were filing issues about their apps not running with musl. However, if it were only an option, and not the default, it might work.

If I recall, Alpine package names aren't that different from their Debian counterparts, so I don't really think that package name migration would be the most annoying part. I think the main challenge would just be managing two different base images, neither having binary compatibility with the other.

At the same time, though, it looks like we already have logic for switching base images thanks to the Ubuntu switch, so it might not even be that big of a problem.

In short, I'm waffling rapidly, but leaning towards this being a possibility.

**Comment by s-h-a-d-o-w** [comment]
> folks were filing issues about their apps not running with musl

That's a huge bummer!
Could that maybe be solved by providing a "completely seperate" (of course you would reuse code under the hood) flavor of nixpacks, like `nixpacks-alpine` or running `nixpacks --build alpine`?
That way, you don't have to do a lot of detection and whatnot. But... if you wanted to, you could maybe analyze the final "build plan" when regular nixpack is used. And if only alpine-compatible providers, etc. are used, you could print a message like "It looks like you may be able to optimize memory usage by using `nixpacks-alpine`."
